### ISSUE TITLE
Support for Nokia 9 B&W DNGs

### ISSUE THREAD

**Issue by Lawrence37** [Beep6581/RawTherapee#6633]
Original report from https://discuss.pixls.us/t/nokia-9-black-and-white-demosaicing-corruption/33944

> Hi. Its related to previous inquiry. Black and white dngs for this camera show up badly. There’s no even settings for demosaicing in this (B&W) mode.
This very issue happened with digiKam too but they could fix it for version 7.3 after reporting to them and libraw (see [431689 – Nokia 9 PureView DNG image demosaicing preview corruption](https://bugs.kde.org/show_bug.cgi?id=431689)).
>
> ![](https://d2x313g9lpht1q.cloudfront.net/original/3X/0/7/0754e695a65242f2106047156fa8ffae261b249e.png)
>
> And rather quickly so I guess shouldn’t be very difficult to solve in RawTherapee as well.
I really thank your work.
I appreciate your help.
Kind regards.
Javier.
>
> I attach the original unmodified DNG
[IMG_20221127_000537.dng](https://discuss.pixls.us/uploads/short-url/u3Lp2rGYoRj9POp2riyFEG1ovA1.dng) (24.6 MB)

[This LibRaw bug report](https://github.com/LibRaw/LibRaw/issues/404) suggests it is an issue decoding uncompressed tiled integer DNGs.

**Comment by Entropy512** [comment]
Hmm, could it be that https://github.com/Beep6581/RawTherapee/issues/6575 also applies to uncompressed?

**Comment by dm30** [comment]
By the way, there are very few options in the RAW tab when editing non-corrupted images*. Does not seem to recognize the sensor, which is, in all 5 cases, the Sony IMX386 I think. 3 units doesn't have any colour filter: only these shoot for black and white.
Also its claimed that there's some sort of fusing of these images (with custom chip) in the final DNG. 
I really appreciate your help and thank you for these great software.

*Only 'RAW white points' and 'Dark Frame'. Nothing more.

**Comment by SimonSegerblomRex** [comment]
> Hmm, could it be that #6575 also applies to uncompressed?

No, this is unrelated. I had a quick look at the file and it's using strips (with 377 rows per strip, which sounds very odd). It's not fun tor try to debug dcraw.cc...

**Comment by dm30** [comment]
Thank you. It was solved in DigiKam, through LibRaw update I think. Perhaps they can give some clue about what was going on.

**Comment by Lawrence37** [comment]
Works with LibRaw enabled.
